### DISCUSSION TITLE
First roadmap

### DISCUSSION THREAD

**Discussion by nelsonmestevao** [coderdojobraga/gongo#8] [Announcements]
1. Update the discord library (urgent)
2. Refactor code to extend discord.Command class (important)
3. Add functionality of giving new belts taking into account restrictions such as: 
   - only mentors can do it, must be in a public chat channel (public in the sense mentors, ninjas and guardians can see it
   - if the belt is not of next level, the bot should warn the user [levels: no belt, white, yellow, blue, green, orange, red, purple, black]
4. Send a resume of the day to an admins channels (must have) 
5. Work on a festive way to celebrate a new belt attribution (See discussion https://github.com/coderdojobraga/gongo/discussions/7) (nice have)
6. Keep the users roles in sync with the API build on [bokken](https://github.com/coderdojobraga/bokken) (future requirement)

**Comment by ruioliveira02** [comment]
I agree with most of it, however I don't think allowing the bot to give new belts is a good idea bmainly because of separation of concerns. I think the online platform should be the only place that could be done.

Regarding 5., refer to my reply in that specific thread.

**Reply by nelsonmestevao** [reply]
The bot is not giving new belts. It is updating the role in users discord bots according to that when an authorized user decides so. Todo so, it must ask bokken API to do it (using a special auth token).

**Reply by ruioliveira02** [reply]
> * only mentors can do it, must be in a public chat channel (public in the sense mentors, ninjas and guardians can see it

I was talking about this in particular. I think synchronization should be one way only (from API to bot). If a mentor wants to changes a belt it should be done via the platform. I just don't think it makes sense being able to do in two different places. Unless of course we don't implement it in the platform and only allow the bot to do it.

**Reply by nelsonmestevao** [reply]
I don't think the bot is very useful if it can't be an interface to update the ninja belts too. Otherwise, this is just a cronjob that syncs user roles with the db. The goal of build an API like bokken is to allow multiple clients, be it a web interface, a discord bot or a mobile app etc.

**Comment by danielsp45** [comment]
Since one of the points of the roadmap is to refactor the code to extend the discord.Command class, can we ditch the docopt library or should we still use it?

**Reply by nelsonmestevao** [reply]
I think we should remove docopt library and use discord.Command fully.
